File: core/src/kr/edoli/imview/ColorDrawable.java
package kr.edoli.imview;

import com.badlogic.gdx.graphics.Color;
import com.badlogic.gdx.graphics.g2d.Batch;
import com.badlogic.gdx.scenes.scene2d.utils.BaseDrawable;
import com.badlogic.gdx.scenes.scene2d.utils.Drawable;

/**
 * Created by 석준 on 2016-02-06.
 */
public class ColorDrawable extends BaseDrawable {

    private Color color;

    public ColorDrawable(Color color) {
        this.color = color;
    }

    @Override
    public void draw(Batch batch, float x, float y, float width, float height) {
        Color preColor = batch.getColor();

        batch.setColor(color);
        batch.draw(Textures.White, x, y, width, height);

        batch.setColor(preColor);

        super.draw(batch, x, y, width, height);
    }
}

File: desktop/src/main/kotlin/kr/edoli/imview/ui/drawable/ColorDrawable.kt
package kr.edoli.imview.ui.drawable

import com.badlogic.gdx.graphics.Color
import com.badlogic.gdx.graphics.g2d.Batch
import com.badlogic.gdx.scenes.scene2d.utils.BaseDrawable
import kr.edoli.imview.ui.UIRes


class ColorDrawable(val color: Color) : BaseDrawable() {

    override fun draw(batch: Batch, x: Float, y: Float, width: Float, height: Float) {
        batch.color = color
        batch.draw(UIRes.white, x, y, width, height)
    }
}